geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/iterators/KryoLazyAggregatingIterator.scala

```diff
@@ -63,18 +65,36 @@ abstract class KryoLazyAggregatingIterator[T <: AnyRef { def isEmpty: Boolean; d
 
     val spec = options(SFT_OPT)
     sft = IteratorCache.sft(spec)
+
     index = try { AccumuloFeatureIndex.index(options(INDEX_OPT)) } catch {
       case NonFatal(e) => throw new RuntimeException(s"Index option not configured correctly: ${options.get(INDEX_OPT)}")
     }
 
-    if (index.serializedWithId) {
+    reusableSf = if (index.serializedWithId) {
```

Comment:
[minor] i kind of don't like pulling this out, as there are side-effects in the if-statement